src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -86,12 +88,14 @@ internal class LocalGrainDirectory :
         internal readonly CounterStatistic UnregistrationsManyRemoteSent;
         internal readonly CounterStatistic UnregistrationsManyRemoteReceived;
 
-        
         public LocalGrainDirectory(Silo silo)
         {
             log = LogManager.GetLogger("Orleans.GrainDirectory.LocalGrainDirectory");
 
             MyAddress = silo.LocalMessageCenter.MyAddress;
+            isSiloNextInTheRing = (siloAddr, hash, excludeMySelf) => (siloAddr.GetConsistentHashCode() <= hash) &&
+                                                                     (!excludeMySelf || !siloAddr.Equals(MyAddress));
```

Comment:
I'm confused? is it really capturing the address. I would expect this code to capture `this` and then accessing the `MyAddress` field, which is no different to calling a proper instance method.